packages/platform-sdk-nano/src/services/identity/public-key.ts

```diff
@@ -15,6 +16,6 @@ export class PublicKey implements Contracts.PublicKey {
 	}
 
 	public async fromSecret(secret: string): Promise<string> {
-		throw new Exceptions.NotSupported(this.constructor.name, "fromSecret");
+		return nanocurrency.derivePublicKey(secret);
```

Comment:
What does the secret look like?

Reply:
I thought this was the private key

Reply:
There's a method called `fromPrivateKey` if you want to implement private key derivation.